### ISSUE TITLE
Clearly document expected behaviour of setRef()

### ISSUE THREAD

**Issue by arturdzm** [eclipse/microprofile-open-api#51]
As a improvement for user experience it should be allowed to provide a short reference name:
This should be clearly defined in specification document

Example:

```
// #/components/parameters/userId
parameter.setRef("userId");

// #/components/responses/NotFound
response.setRef("NotFound");

// user also can provides a full reference
response.setRef("#/components/responses/NotFound");
```

**Comment by arthurdm** [comment]
API javadoc change

**Comment by turkeylurkey** [comment]
I'll investigate this.

**Comment by arthurdm** [comment]
Closed via PR #104
